WooCommerce/Classes/ViewRelated/Filters/FilterListViewController.swift

```diff
@@ -348,29 +349,64 @@ private extension FilterListViewController {
 
         let data: [FilterType]
 
+        /// Parent view controller. Used to launch the promoted url web view.
+        ///
+        private weak var hostViewController: UIViewController?
+
         private let onItemSelectedSubject = PassthroughSubject<FilterType, Never>()
         var onItemSelected: AnyPublisher<FilterType, Never> {
             onItemSelectedSubject.eraseToAnyPublisher()
         }
 
-        init(navigationBarTitle: String, data: [FilterType], selected: FilterType) {
+        init(navigationBarTitle: String, data: [FilterType], selected: FilterType, hostViewController: UIViewController? = nil) {
             self.navigationBarTitle = navigationBarTitle
             self.data = data
             self.selected = selected
+            self.hostViewController = hostViewController
         }
 
         func isSelected(model: FilterType) -> Bool {
             selected?.description == model.description
         }
 
         func handleSelectedChange(selected: FilterType, viewController: ViewController) {
+            // Do not allow selection for an unavailable promotable type.
+            if let promotable = selected as? PromotableProductType, !promotable.isAvailable {
+                return
+            }
+
             onItemSelectedSubject.send(selected)
             self.selected = selected
         }
 
         func configureCell(cell: BasicTableViewCell, model: FilterType) {
             cell.textLabel?.text = model.description
             cell.accessibilityIdentifier = model.description
+            cell.accessoryView = nil
+
+            if let promotable = model as? PromotableProductType, !promotable.isAvailable {
+                cell.accessoryView = createPromoteButton(promotableType: promotable)
+            }
```

Comment:
It'd be great to see `FilterListViewController` decoupled from product filter-specific logic. I can also help look deeper into the feasibility later, what do you think about:
- Adding an optional `cellConfiguration: ((UITableViewCell) -> Void)?` callback to allow the filter model to configure a cell's appearance like accessory view
- Creating a new type similar to `FilterType` like `PromotableFilterType` with the necessary promotion data (title and URL), then adding an enum case to `FilterListValueSelectorConfig` like `promotableStaticOptions(options: [PromotableFilterType])`

Reply:
I tried both of these and none convinced me entirely. The second even implied a massive(I think unnecessary) amount of diff changes 😅 

As this seemed like a one-time change in a screen that is not very prone to change I think this is fine for now. 
But if we start expanding this further then we should consider a more extensible approach.